service/s3/internal/customizations/update_endpoint.go

```diff
@@ -7,68 +7,120 @@ import (
 	"net/url"
 	"strings"
 
+	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/awslabs/smithy-go/middleware"
 	smithyhttp "github.com/awslabs/smithy-go/transport/http"
 
+	awsmiddleware "github.com/aws/aws-sdk-go-v2/aws/middleware"
 	"github.com/aws/aws-sdk-go-v2/service/internal/s3shared"
+
+	internalendpoints "github.com/aws/aws-sdk-go-v2/service/s3/internal/endpoints"
 )
 
-// UpdateEndpointOptions provides the options for the UpdateEndpoint middleware setup.
-type UpdateEndpointOptions struct {
-	// region used
-	Region string
+// EndpointResolver interface for resolving service endpoints.
+type EndpointResolver interface {
+	ResolveEndpoint(region string, options EndpointResolverOptions) (aws.Endpoint, error)
+}
 
+// EndpointResolverOptions is the service endpoint resolver options
+type EndpointResolverOptions = internalendpoints.Options
+
+// UpdateEndpointParameterAccessor represents accessor functions used by the middleware
+type UpdateEndpointParameterAccessor struct {
 	// functional pointer to fetch bucket name from provided input.
 	// The function is intended to take an input value, and
 	// return a string pointer to value of string, and bool if
 	// input has no bucket member.
 	GetBucketFromInput func(interface{}) (*string, bool)
 
+	// indicates if an operation supports s3 transfer acceleration.
+	SupportsAccelerate bool
+}
```

Comment:
Should `SupportsAccelerate` be a member of the `UpdateEndpointOptions`? Seems out of place in the Accessor's group.